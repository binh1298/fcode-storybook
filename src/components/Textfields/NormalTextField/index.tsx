import { InputAdornment } from "@material-ui/core";
import TextFieldBase, { TexfieldBaseProps } from "src/components/Textfields/TextFieldBase";

export interface NormalTextFieldProps extends TexfieldBaseProps {
    icon?: React.ReactNode;
}

const NormalTextField = (props: NormalTextFieldProps) => {
    const { icon } = props;
    return (
        <TextFieldBase
            {...props}
            InputProps={{
                startAdornment: <InputAdornment position="start">{icon}</InputAdornment>,
                style: { fontSize: "1rem" },
            }}
            InputLabelProps={{ style: { fontSize: "1rem" } }}
        />
    );
};

export default NormalTextField;
