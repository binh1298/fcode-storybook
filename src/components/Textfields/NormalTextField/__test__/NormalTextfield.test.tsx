import NormalTextField, { NormalTextFieldProps } from "../";

import { render, RenderResult } from "@testing-library/react";

describe("Structure of <TextfieldBase />", () => {
    const props: NormalTextFieldProps = {
        id: "test",
        label: "Test",
        value: "Value",
    };

    it("should have the label for the input", () => {
        const { getByRole } = render(<NormalTextField {...props} />);
        expect(getByRole("textbox", { name: "Test" })).not.toBeUndefined();
    });
});

describe("<NormalTextField />", () => {
    let wrapper: RenderResult;

    const propsWithErrorMessage: NormalTextFieldProps = {
        label: "Test",
        value: "Value",
        errorMessage: "error message",
    };

    it("should show error color when errorMessage is available", () => {
        wrapper = render(<NormalTextField {...propsWithErrorMessage} />);
        expect(wrapper.container).toMatchSnapshot();
    });
});

describe("<NormalTextField />", () => {
    let wrapper: RenderResult;

    const propsWithNoErrorMessage: NormalTextFieldProps = {
        label: "Test",
        value: "Value",
        errorMessage: undefined,
    };

    it("should not show error color when not have any error message", () => {
        wrapper = render(<NormalTextField {...propsWithNoErrorMessage} />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
