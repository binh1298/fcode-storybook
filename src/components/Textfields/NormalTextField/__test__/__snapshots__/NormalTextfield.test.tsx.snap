// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NormalTextField /> should not show error color when not have any error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-17 makeStyles-root-21"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root makeStyles-cssLabel-15 makeStyles-cssLabel-19 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      style="font-size: 1rem;"
    >
      Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-fontColor-16 makeStyles-fontColor-20 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
      style="font-size: 1rem;"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart"
      />
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
        type="text"
        value="Value"
      />
    </div>
  </div>
</div>
`;

exports[`<NormalTextField /> should show error color when errorMessage is available 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-10 makeStyles-root-14"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root makeStyles-cssLabel-8 makeStyles-cssLabel-12 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-error Mui-error MuiFormLabel-filled"
      data-shrink="true"
      style="font-size: 1rem;"
    >
      Test
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-fontColor-9 makeStyles-fontColor-13 Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
      style="font-size: 1rem;"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart"
      />
      <input
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
        type="text"
        value="Value"
      />
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-filled"
    >
      error message
    </p>
  </div>
</div>
`;
